packages/contract/lib/utils/index.js

```diff
@@ -21,23 +21,26 @@ const Utils = {
     return web3Utils.isBN(val) || web3Utils.isBigNumber(val);
   },
 
-  is_tx_params(val) {
+  isTxParams(val) {
     if (!Utils.is_object(val)) return false;
     if (Utils.is_big_number(val)) return false;
 
-    const allowed_fields = {
-      from: true,
-      to: true,
-      gas: true,
-      gasPrice: true,
-      value: true,
-      data: true,
-      nonce: true,
-      privateFor: true
-    };
-
-    for (let field_name of Object.keys(val)) {
-      if (allowed_fields[field_name]) return true;
+    const allowedFields = new Set([
+      "from",
+      "to",
+      "gas",
+      "gasPrice",
+      "value",
+      "data",
+      "nonce",
+      "privateFor",
+      "overwrite"
```

Comment:
I'm lacking some context here, so please bear with me :-)

What are `privateFor` and `overwrite` used for (since these aren't valid RPC properties)?

And should aliases like `input` (`data`) and `gasLimit` (`gas`) be considered? Also, `r`, `s`, and ,`v` are present in _raw_ transactions (though over JSON-RPC they would be serialized via RLP encoding, not as raw JSON)

Reply:
I'm not @eggplantzzz, but I know `privateFor` is a quorum thing; no idea about `overwrite`.

Aliases of course would require not just allowing them but programming them to work, so that would be a bigger change...

Reply:
Huh, looks like `overwrite` is specific to the deployer and signals it to do a reset?

Reply:
Yeah, `overwrite` is Truffle-specific. You can find info [here](https://www.trufflesuite.com/docs/truffle/getting-started/running-migrations#deployer-deploy-contract-args-options-).

Reply:
@haltman-at i'm not sure where this stuff ends up, but the aliases _do_ work in ganache and other ethereum nodes.

Reply:
@davidmurdoch Interesting, I had no idea! :)